benches/benches/block_target_gas_set/contract.rs

```diff
@@ -1,20 +1,741 @@
+        let instructions = call_contract_repeat();
+        replace_contract_in_service(&mut service, &contract_id, contract);
+        run_with_service(
+            "contract/ret contract",
+            group,
+            instructions,
+            script_data.clone(),
+            &service,
+            contract_id,
+            &rt,
+            &mut rng,
+        );
+    }
+
+    // retd contract
+    {
+        for i in arb_dependent_cost_values() {
+            let contract = vec![op::movi(0x14, i), op::retd(RegId::ONE, 0x14)];
+            let instructions = call_contract_repeat();
+            replace_contract_in_service(&mut service, &contract_id, contract);
+            let id = format!("contract/retd contract {:?}", i);
+            run_with_service(
+                &id,
+                group,
+                instructions,
+                script_data.clone(),
+                &service,
+                contract_id,
+                &rt,
+                &mut rng,
+            );
+        }
+    }
+
+    //     run_group_ref(
+    //         &mut c.benchmark_group("rvrt_contract"),
+    //         "rvrt_contract",
+    //         VmBench::contract(rng, op::ret(RegId::ONE)).unwrap(),
+    //     );
+
+    // TODO: Is `rvrt` even possible to bench?
+    // {
+    //     let contract = vec![op::rvrt(RegId::ONE)];
+    //     let instructions = call_contract_repeat();
+    //     replace_contract_in_service(&mut service, &contract_id, contract);
+    //     run_with_service(
+    //         "contract/rvrt contract",
     //         group,
-    //         vec![
-    //             op::movi(0x10, i),
-    //             op::retd(RegId::ONE, 0x10),
-    //             op::jmpb(RegId::ZERO, 0),
-    //         ]
-    //         .to_vec(),
+    //         call_contract_repeat(),
+    //         script_data.clone(),
+    //         &service,
+    //         contract_id,
+    //         &rt,
+    //         &mut rng,
+    //     );
+    // }
+
+    // smo
+
+    {
+        let predicate = op::ret(RegId::ONE).to_bytes().to_vec();
+        let owner = Input::predicate_owner(&predicate);
+        let coin_input = Input::coin_predicate(
+            Default::default(),
+            owner,
+            u32::MAX as Word,
+            AssetId::zeroed(),
+            Default::default(),
+            Default::default(),
+            Default::default(),
+            predicate,
+            vec![],
+        );
+        let extra_inputs = vec![coin_input];
+        for i in arb_dependent_cost_values() {
+            let contract = vec![
+                op::gtf_args(0x15, 0x00, GTFArgs::ScriptData),
+                // Offset 32 + 8 + 8 + 32
+                op::addi(0x15, 0x15, 32 + 8 + 8 + 32), // target address pointer
+                op::addi(0x16, 0x15, 32),              // data ppinter
+                op::movi(0x17, i),                     // data length
+                op::smo(0x15, 0x16, 0x17, 0x18),
+                op::ret(RegId::ZERO),
```

Comment:
You can pass a lot of the assets during the first call and jump backward here. Also, you can create a message with 1 coins instead of 10